block-streamer/src/receiver.rs

```diff
@@ -106,184 +66,39 @@ mod tests {
         }
     }
 
-    #[test]
-    pub fn test_receive_chunk() {
-        let mut harness = TestHarness::new();
-
-        let chunk = TransmissionChunk {
-            cid_marker: vec![],
-            chunk_offset: 0,
-            data: vec![],
-        };
-        harness.receiver.handle_chunk_msg(chunk.clone()).unwrap();
-        let entry = harness.receiver.cid_chunks.first_key_value().unwrap();
-        assert_eq!(entry.0, &chunk.cid_marker);
-        assert!(entry.1.contains(&chunk));
-    }
-
-    #[test]
-    pub fn test_receive_cid() {
-        let mut harness = TestHarness::new();
-        let cid = b"101010101010101".to_vec();
-        harness.receiver.handle_cid_msg(cid.clone()).unwrap();
-
-        assert_eq!(
-            harness.receiver.cids_to_build.first_key_value().unwrap().1,
-            &cid
-        );
-    }
-
     #[tokio::test]
-    pub async fn test_child_block_assembly() {
+    pub async fn test_receive_block_msg() {
         let mut harness = TestHarness::new();
-        let h = Code::Sha2_256.digest(b"test this");
-        let cid = Cid::new_v1(0x55, h);
+        let cid = Cid::new_v1(0x55, cid::multihash::Code::Sha2_256.digest(b"00"));
 
-        let wrapper = BlockWrapper {
+        let block_msg = TransmissionMessage::Block(TransmissionBlock {
             cid: cid.to_bytes(),
-            payload: BlockPayload {
-                data: b"11111".to_vec(),
-                links: vec![],
-            },
-        };
-        let chunks = wrapper.to_chunks().unwrap();
-        for c in chunks {
-            harness.receiver.handle_transmission_msg(c).await.unwrap();
-        }
+            data: b"1871217171".to_vec(),
+            links: vec![cid.to_bytes()],
+        });
 
-        harness.receiver.attempt_block_assembly().unwrap();
-        assert_eq!(harness.storage.list_available_cids().unwrap().len(), 1);
+        let res = harness.receiver.handle_transmission_msg(block_msg).await;
+        assert!(res.is_ok());
```

Comment:
It might be a nice bonus to assert the `res` values, but not critical.

Reply:
I would agree if `res` had some meaningful value, but in this case it's a unit `()`, so I don't think we would gain much by asserting that vs on `is_ok`.

Reply:
Ok, I think I understand, you're just testing for a non-error condition, there's no state to validate after the transmission message is handled.